arwen/contexts/runtime.go

```diff
@@ -66,17 +72,36 @@ func (context *runtimeContext) InitStateFromContractCallInput(input *vmcommon.Co
 
 func (context *runtimeContext) PushState() {
 	newState := &runtimeContext{
-		vmInput:      context.vmInput,
-		scAddress:    context.scAddress,
-		callFunction: context.callFunction,
-		readOnly:     context.readOnly,
+		vmInput:       context.vmInput,
+		scAddress:     context.scAddress,
+		callFunction:  context.callFunction,
+		readOnly:      context.readOnly,
+		asyncCallInfo: context.asyncCallInfo,
 	}
 
 	context.stateStack = append(context.stateStack, newState)
 }
 
-func (context *runtimeContext) PushInstance() {
-	context.instanceStack = append(context.instanceStack, context.instance)
+func (runtime *runtimeContext) PopState() error {
```

Comment:
i think popState should not return in error. Push and Pop always work in syncron.

Reply:
True, `Pop` will only fail if there's a coding error in Arwen and the users cannot take any action for that case. Removed.